lib/mongodb.js

```diff
@@ -1237,8 +1256,7 @@ MongoDB.prototype.all = function all(model, filter, options, callback) {
   fields = self.fromPropertyToDatabaseNames(model, fields);
 
   if (fields) {
-    // convert the array of fields into a projection object see http://mongodb.github.io/node-mongodb-native/3.0/api/Collection.html#find
-    var findOpts = {projection: fields};
+    var findOpts = {projection: Array.isArray(fields) ? fieldsArrayToObj(fields) : fields};
```

Comment:
This can be simplified to `var findOpts = {projection: fieldsArrayToObj(fields)};` as `fireldsArrayToObj()` has a check to return as is for a non-array type.

Reply:
Done